api/v1beta1/glance_webhook.go

```diff
@@ -76,8 +110,16 @@ var _ webhook.Validator = &Glance{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Glance) ValidateCreate() error {
 	glancelog.Info("validate create", "name", r.Name)
+	// At creation time, if the CR has an invalid keystoneEndpoint value (that
+	// doesn't match with any defined backend), return an error.
+	if !r.Spec.isValidKeystoneEP() {
+		return errors.New("KeystoneEndpoint is assigned to an invalid glanceAPI instance")
+	}
 
-	// TODO(user): fill in your validation logic upon object creation.
+	//TODO:
+	// - Two glanceAPI with the same name can't be deployed
```

Comment:
Is it even possible to deploy two `GlanceAPI`s with the same name, given that they are specified in a map and the keys are the names?

Reply:
No, it's actually not possible, and the CR is not updated, however, I'd like to explicitly deny specific actions like this one, and return an error using `webhooks`, which is something we do for both the `keystoneEndpoint` and  `layout` changes

Reply:
This seems to be covered at creation time:

Example:

```
apiVersion: glance.openstack.org/v1beta1
kind: Glance
...
spec:
  serviceUser: glance
  databaseInstance: openstack
  databaseUser: glance
  glanceAPIs:
    api0:
      type: "single"
      preserveJobs: false
      replicas: 1
    api0:
      type: "single"
      preserveJobs: false
      replicas: 1
    api1:
      type: "single"
      preserveJobs: false
      replicas: 1

```

results in:

```
[stack@osp-storage-04 layout]$ oc kustomize single/ | oc create -f -
error: map[string]interface {}(nil): yaml: unmarshal errors:
  line 23: mapping key "api0" already defined at line 17
error: no objects passed to create
```

If I simply `oc create glance-cr.yaml` the deploy starts and it results in the last definition taking place, because it's a `map[string]GlanceAPITemplate`, and it's like you override the value for an existing key.

```
glance-api0-single-0        3/3     Running     0          105s
glance-api1-single-0        3/3     Running     0          105s
```

Sorry for not catching this in my previous comment, but basically by definition `maps` don't allow duplicated keys, even if they're strings, which means that if you define the same API multiple times, it's just an override of the previous definitions.